modules/navier_stokes/src/auxkernels/NSPressureAux.C

```diff
@@ -14,35 +19,23 @@ InputParameters validParams<NSPressureAux>()
   InputParameters params = validParams<AuxKernel>();
 
   // Mark variables as required
-  params.addRequiredCoupledVar("rho", "");
-  params.addRequiredCoupledVar("u", "x-velocity");
-  params.addCoupledVar("v", "y-velocity"); // Only required in >= 2D
-  params.addCoupledVar("w", "z-velocity"); // Only required in 3D...
-  params.addRequiredCoupledVar("rhoE", "total energy");
-
-  // Parameters with default values
-  params.addRequiredParam<Real>("gamma", "Ratio of specific heats");
+  params.addRequiredCoupledVar("specific_volume", "");
+  params.addRequiredCoupledVar("internal_energy", "");
+  params.addRequiredParam<UserObjectName>("fluid_properties", "The name of the user object for fluid properties");
 
   return params;
 }
 
 NSPressureAux::NSPressureAux(const InputParameters & parameters) :
     AuxKernel(parameters),
-    _rho(coupledValue("rho")),
-    _u_vel(coupledValue("u")),
-    _v_vel(_mesh.dimension() >= 2 ? coupledValue("v") : _zero),
-    _w_vel(_mesh.dimension() == 3 ? coupledValue("w") : _zero),
-    _rhoE(coupledValue("rhoE")),
-    _gamma(getParam<Real>("gamma")) // can't use Material properties in Nodal Aux...
+    _specific_volume(coupledValue("specific_volume")),
+    _internal_energy(coupledValue("internal_energy")),
+    _fp(getUserObject<IdealGasFluidProperties>("fluid_properties"))
```

Comment:
Do you want to restrict this to just ideal gas?  You should be able to use `SinglePhaseFluidProperties` here which would give you any fluid properties... Similarly for `NSMachAux`.